AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -0,0 +1,352 @@
+
+
+#import "ASVideoNode.h"
+
+@interface ASVideoNode ()
+{
+  ASDN::RecursiveMutex _lock;
+  
+  __weak id<ASVideoNodeDelegate> _delegate;
+
+  BOOL _shouldBePlaying;
+  AVAsset *_asset;
+  
+  AVPlayerItem *_currentItem;
+  AVPlayer *_player;
+  
+  ASButtonNode *_playButton;
+  ASDisplayNode *_playerNode;
+  ASDisplayNode *_spinner;
+  ASVideoGravity _gravity;
+}
+
+@end
+
+@interface ASDisplayNode ()
+- (void)setInterfaceState:(ASInterfaceState)newState;
```

Comment:
Can now remove this entire forward declaration.  interfaceStateDidChange:forState: may still be in the +Beta header but is safe to use.